main/src/main/java/org/mobilitydata/gtfsvalidator/validator/UrlConsistencyValidator.java

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright 2021 MobilityData IO
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import java.util.HashMap;
+import java.util.Map;
+import javax.inject.Inject;
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidator;
+import org.mobilitydata.gtfsvalidator.annotation.SchemaExport;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.SeverityLevel;
+import org.mobilitydata.gtfsvalidator.notice.ValidationNotice;
+import org.mobilitydata.gtfsvalidator.table.GtfsAgency;
+import org.mobilitydata.gtfsvalidator.table.GtfsAgencyTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsRoute;
+import org.mobilitydata.gtfsvalidator.table.GtfsRouteTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsStop;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTableContainer;
+
+/**
+ * Validates that:
+ * <ul>
+ *  <li>all {@code GtfsRoute} have: {@code routes.route_url != agency.agency_url}.</li>
+ *  <li>all {@code GtfsStop} have: {@code stops.route_url != agency.agency_url}.</li>
+ *  <li>all {@code GtfsStop} have: {@code stops.route_url != routes.route_url}.</li>
+ * </ul>
+ *
+ * <p>Generated notice:
+ *
+ * <ul>
+ *   <li>{@link SameRouteAndAgencyUrlNotice} - for a record from "routes.txt", the value of
+ *   {@code routes.route_url} is shared by a record from "agency.txt".
+ *   <li>{@link SameStopAndAgencyUrlNotice} - for a record from "stops.txt", the value of
+ *   {@code stops.route_url} is shared by a record from "agency.txt".
+ *   *   <li>{@link SameStopAndRouteUrlNotice} - for a record from "stops.txt", the value of
+ *   {@code stops.route_url} is shared by a record from "routes.txt".
+ * </ul>
```

Comment:
I suggest to remove this section because notice classes are documented below in this file.

If you want to keep this section, please fix the typo: `stops.stop_url`.